### ISSUE TITLE
Can we get an update from these people

### ISSUE THREAD

**Issue by def14nt** [selfagency/microsoft-drop-ice#350]
If you can **not** leave GitHub, send a PR that removes your name from the list immediately, so we can keep track of the people that are *actually* going to leave this platform.

Please take action, because detainment, denaturalization, deportation, and family separation are *very* serious issues, and we need to see some action from the people that have signed this. How do we stand up to a monopoly like Microsoft when we can't put our heads together?

I have taken #111 into consideration, and seeing that GitLab is affiliated with the US Air Force, is moving to them a bad idea? I think we should take self-hosted Git services into consideration, like [Gitea](https://gitea.io/en-us/), [a self-hosted instance of GitLab](https://gitlab.com/gitlab-org/gitlab-ce) or [GitBucket, which is also self hosted](https://gitbucket.github.io/).

## Migration from GitHub

Self hosted looks like the way to go, because it leaves a *lot* less control in the hands of companies, so they can't just "go bad" on you and install a paywall, or provide "better integration" between services.

By using GitHub and investing our time into it, we are making it harder for us to leave. Think of it like quicksand. The earlier you leave GitHub, the easier it'll be for you, and it leaves Microsoft with less control over your development workflow as a whole. 

*Find more self-hosted solutions in this subreddit, as I have probably missed a few: https://www.reddit.com/r/selfhosted/comments/6aahi0/about_selfhosted_git_server/*

Below are some links for migrating away from GitHub.

+ [How to migrate from GitHub to GitLab](https://docs.gitlab.com/ee/user/project/import/github.html).
+ [A blog item about why Microsoft acquiring GitHub is bad news](https://dev.to/maxart2501/what-i-think-is-bad-about-microsoft-acquiring-github-gme).
+ [An update on Microsoft buying GitHub](https://www.theverge.com/2018/6/18/17474284/microsoft-github-acquisition-developer-reaction).
+ [Exporting your data from GitHub](https://github.blog/2018-12-19-download-your-data/).
+ [Deleting your GitHub account](https://help.github.com/articles/deleting-your-user-account/).
+ [Migrating to a self hosted GitLab instance 👍](https://docs.gitlab.com/ee/user/project/import/).

People don't trust Microsoft, and they are not hiding it. If you disagree with this acquisition, feel free to share this repository on social media, because it really does have some potential.

I hope this post has inspired / helped you to leave GitHub in any way, as that is the end goal of this project.

With this in mind, I'd like to thank everyone that has signed this petition, but it's still not enough. We need to show Microsoft we're not going to take this.

Thank you for your time, and have a great weekend.

**Comment by rbstrachan** [comment]
Thank you for the update. If you can spare some time, I would like to clarify a few things.

**First**
You said,
> If you can **not** leave GitHub, send a PR that removes your name from the list immediately [...]

So far, I have migrated all of my repos to GitLab and I am no longer active on GitHub. I am however hesitant to delete my GitHub account. Is this grounds for removing my name from the list?

**Second**
You mentioned that GitLab is affiliated with the USAF and therefore may not be the best choice to migrate to. I am, unfortunately, not willing to migrate anywhere else. Is *this* reason enough to remove my name from the list?

---

You are right. Family separation is serious, and with the recent developments with the Trump Administration unable and unwilling to locate hundreds of children, it becomes an even greater concern.
I signed this list because I wanted to see change and I had the opportunity to be a part of it. As time goes on, however, I don't feel like I can bring about the change that I envisioned. The last thing I want to do is inhibit the efforts of others who have the ability to make a lasting impression on Microsoft. My question is, is it time for me to walk away from this movement?

Thank you for your help.

**Comment by def14nt** [comment]
@rbstrachan I mentioned that it may not be the best because it supports the USAF, but I think that's a fight for another day, and nothing to do with this movement.

I do not think that using Gitlab is reason to remove your name from the list, but please do not feel restricted. If you do not want to be a part of this movement, send a PR.

Whether or not you walk away from this movement is purely up to you :+1:

**Comment by def14nt** [comment]
@selfagency Should we start pinging people that have signed up for this? I'm worried that might discount us, and people might instead see us as a bother (who says they don't already judging by their attitudes).

**Comment by randolf** [comment]
I think that a major problem with removing names from the list of signatories is that it will create the impression that people are changing their minds about supporting it.

A better solution may be to add an asterisk beside the names of those who have moved away from GitHub to denote that they've done so (or place an asterisk beside those who have not).

Also important to not lose perspective on is that some people may be required to continue to use GitHub because their employer relies on it, and said employer has a different view on the matter.  I think that it's important to be careful about not excluding these people (or anyone) from participating in this petition.

Overall, having more signatures is always better because it shows that this petition has more support.  Please don't remove any signatures.

**Comment by randolf** [comment]
I agree that self-hosted is a much better choice for the reasons you stated (and I agree with your reason).

Of course, the problem with self-hosted options is that they tend to have a greater potential to disappear because hosting has financial, physical space, and quality of life burdens that make well-funded and professionally-managed systems that are publicly-accessible very attractive.

Perhaps if there was a way that self-hosted solutions could work to replicate/mirror the works of others in some automated fashion?  Or does GitLab's self-hosted solution already offer this functionality?

**Comment by def0x099** [comment]
@randolf The `*` system sounds good. How would we check if they have left GitHub, though?

**Comment by badrihippo** [comment]
One thing's still not clear: what does "leaving GitHub" mean in this context? I can see it as (a) using something else as your primary repo host but coming back here once in a while to dip into projects, (b) using some other host keeping this account open but inactive, and (c) actually deleting your account forever.

I'm probably between categories (a) and (b), because I meant to leave entirely but then got pulled into a project which is hosted on GitHub (am yet to convince them to move).

**Comment by randolf** [comment]
> @randolf The `*` system sounds good. How would we check if they have left GitHub, though?

I suppose it should be fairly easy to check if a given user's GitHub account has been deleted, but I think a better approach is to work on replication between self-hosted repositories.  As I understand it, GitLab uses PostgreSQL, which has some excellent replication options, including read-only replication that could be useful for sites that are only interested in mirroring certain projects.

**Comment by joshua-s** [comment]
I would very much like to support this, but must use GitHub for my employer. I will gladly move all other projects, however. Should I rescind my PR?
